# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/ProfilerHelpers.h
# Commit: 340dfa69aed8
# Full Hash: 340dfa69aed87f29b9fdeef59a650ad5eb8c1b7b
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2019-09-30 16:21:37
# Regressor Bug: 1168606
# File Overlap Count: 1
# Description:
#   Bug 1168606 - Remove duplication in uses of IDB_LOG_MARK. r=ttung,asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D46576
# ==============================================================================

diff -r a864ed2ae59a -r 340dfa69aed8 dom/indexedDB/ProfilerHelpers.h
--- a/dom/indexedDB/ProfilerHelpers.h	Mon Sep 30 08:54:17 2019 +0000
+++ b/dom/indexedDB/ProfilerHelpers.h	Fri Sep 27 10:11:45 2019 +0000
@@ -290,7 +290,7 @@
 }  // namespace dom
 }  // namespace mozilla
 
-#define IDB_LOG_MARK(_detailedFmt, _conciseFmt, ...)                           \
+#define IDB_LOG_MARK2(_detailedFmt, _conciseFmt, ...)                          \
   do {                                                                         \
     using namespace mozilla::dom::indexedDB;                                   \
                                                                                \
@@ -317,10 +317,84 @@
     }                                                                          \
   } while (0)
 
+#define IDB_LOG_MARK(_detailedFmt, _conciseFmt, _loggingId, ...)             \
+  IDB_LOG_MARK2("IndexedDB %s: " _detailedFmt, "IndexedDB %s: " _conciseFmt, \
+                _loggingId, ##__VA_ARGS__)
+
 #define IDB_LOG_ID_STRING(...) \
   mozilla::dom::indexedDB::LoggingIdString(__VA_ARGS__).get()
 
 #define IDB_LOG_STRINGIFY(...) \
   mozilla::dom::indexedDB::LoggingString(__VA_ARGS__).get()
 
+// IDB_LOG_MARK_DETAILED_PARENT and IDB_LOG_MARK_DETAILED_CHILD should have the
+// same width.
+#define IDB_LOG_MARK_DETAILED_PARENT "Parent"
+#define IDB_LOG_MARK_DETAILED_CHILD "Child "
+
+#define IDB_LOG_MARK_CONCISE_PARENT "P"
+#define IDB_LOG_MARK_CONCISE_CHILD "C"
+
+#define IDB_LOG_MARK_DETAILED_TRANSACTION "Transaction[%lld]"
+#define IDB_LOG_MARK_DETAILED_REQUEST "Request[%llu]"
+
+#define IDB_LOG_MARK_CONCISE_TRANSACTION "T[%lld]"
+#define IDB_LOG_MARK_CONCISE_REQUEST "R[%llu]"
+
+#define IDB_LOG_MARK_TRANSACTION_REQUEST(                                      \
+    _detailedPeer, _concisePeer, _detailedFmt, _conciseFmt, _loggingId,        \
+    _transactionSerialNumber, _requestSerialNumber, ...)                       \
+  IDB_LOG_MARK(_detailedPeer " " IDB_LOG_MARK_DETAILED_TRANSACTION             \
+                             " " IDB_LOG_MARK_DETAILED_REQUEST                 \
+                             ": " _detailedFmt,                                \
+               _concisePeer " " IDB_LOG_MARK_CONCISE_TRANSACTION               \
+                            " " IDB_LOG_MARK_CONCISE_REQUEST ": " _conciseFmt, \
+               _loggingId, _transactionSerialNumber, _requestSerialNumber,     \
+               ##__VA_ARGS__)
+
+#define IDB_LOG_MARK_PARENT_TRANSACTION_REQUEST(                               \
+    _detailedFmt, _conciseFmt, _loggingId, _transactionSerialNumber,           \
+    _requestSerialNumber, ...)                                                 \
+  IDB_LOG_MARK_TRANSACTION_REQUEST(                                            \
+      IDB_LOG_MARK_DETAILED_PARENT, IDB_LOG_MARK_CONCISE_PARENT, _detailedFmt, \
+      _conciseFmt, _loggingId, _transactionSerialNumber, _requestSerialNumber, \
+      ##__VA_ARGS__)
+
+#define IDB_LOG_MARK_CHILD_TRANSACTION_REQUEST(_detailedFmt, _conciseFmt,    \
+                                               _transactionSerialNumber,     \
+                                               _requestSerialNumber, ...)    \
+  IDB_LOG_MARK_TRANSACTION_REQUEST(                                          \
+      IDB_LOG_MARK_DETAILED_CHILD, IDB_LOG_MARK_CONCISE_CHILD, _detailedFmt, \
+      _conciseFmt, IDB_LOG_ID_STRING(), _transactionSerialNumber,            \
+      _requestSerialNumber, ##__VA_ARGS__)
+
+#define IDB_LOG_MARK_CHILD_REQUEST(_detailedFmt, _conciseFmt,                \
+                                   _requestSerialNumber, ...)                \
+  IDB_LOG_MARK(IDB_LOG_MARK_DETAILED_CHILD " " IDB_LOG_MARK_DETAILED_REQUEST \
+                                           ": " _detailedFmt,                \
+               IDB_LOG_MARK_CONCISE_CHILD " " IDB_LOG_MARK_CONCISE_REQUEST   \
+                                          ": " _conciseFmt,                  \
+               IDB_LOG_ID_STRING(), _requestSerialNumber, ##__VA_ARGS__)
+
+#define IDB_LOG_MARK_TRANSACTION(_detailedPeer, _concisePeer, _detailedFmt,  \
+                                 _conciseFmt, _loggingId,                    \
+                                 _transactionSerialNumber, ...)              \
+  IDB_LOG_MARK(                                                              \
+      _detailedPeer " " IDB_LOG_MARK_DETAILED_TRANSACTION ": " _detailedFmt, \
+      _concisePeer " " IDB_LOG_MARK_CONCISE_TRANSACTION ": " _conciseFmt,    \
+      _loggingId, _transactionSerialNumber, ##__VA_ARGS__)
+
+#define IDB_LOG_MARK_PARENT_TRANSACTION(_detailedFmt, _conciseFmt, _loggingId, \
+                                        _transactionSerialNumber, ...)         \
+  IDB_LOG_MARK_TRANSACTION(                                                    \
+      IDB_LOG_MARK_DETAILED_PARENT, IDB_LOG_MARK_CONCISE_PARENT, _detailedFmt, \
+      _conciseFmt, _loggingId, _transactionSerialNumber, ##__VA_ARGS__)
+
+#define IDB_LOG_MARK_CHILD_TRANSACTION(_detailedFmt, _conciseFmt,     \
+                                       _transactionSerialNumber, ...) \
+  IDB_LOG_MARK_TRANSACTION(IDB_LOG_MARK_DETAILED_CHILD,               \
+                           IDB_LOG_MARK_CONCISE_CHILD, _detailedFmt,  \
+                           _conciseFmt, IDB_LOG_ID_STRING(),          \
+                           _transactionSerialNumber, ##__VA_ARGS__)
+
 #endif  // mozilla_dom_indexeddb_profilerhelpers_h__
